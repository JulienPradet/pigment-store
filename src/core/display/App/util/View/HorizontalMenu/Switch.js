import React from 'react'

export default class Switch extends React.Component {
  constructor () {
    super()
    this.onChange = this.onChange.bind(this)
  }

  onChange (event) {
    this.props.onChange(event.target.checked)
  }

  render () {
    return <div>
      <input type='checkbox' checked={this.props.checked} onChange={this.onChange} />
    </div>
  }
}
