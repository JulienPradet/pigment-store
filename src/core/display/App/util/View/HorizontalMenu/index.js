import React from 'react'
import styles from './horizontalmenu.css'

export const Container = ({children}) => <div className={styles.container}>{children}</div>
Container.displayName = 'HorizontalMenu.Container'

export const Group = ({children}) => <div className={styles.group}>{children}</div>
Group.displayName = 'HorizontalMenu.Group'

export const Item = ({children}) => <div className={styles.item}>{children}</div>
Item.displayName = 'HorizontalMenu.Item'
