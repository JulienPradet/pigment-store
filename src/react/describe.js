import describe from './definition/Component'

export default describe
