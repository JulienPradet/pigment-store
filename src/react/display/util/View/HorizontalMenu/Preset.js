import React from 'react'
import classnames from 'classnames'
import styles from './horizontalmenu.css'

export const PresetList = ({presets, onSelect, selected}) => <div className={styles.presetList}>
  {presets.map((preset, index) => <Preset key={index} active={preset.preset === selected} preset={preset} onSelect={onSelect} />)}
</div>

export const Preset = ({active, preset, onSelect}) => <div className={classnames(styles.preset, {[styles.presetActive]: active})}>
  <button onClick={() => onSelect(preset.preset)}>{preset.label}</button>
</div>
