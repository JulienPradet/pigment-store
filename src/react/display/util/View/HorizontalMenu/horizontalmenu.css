.container {
  display: flex;
  justify-content: space-between;
  align-items: stretch;
  background: #fff;
  padding: 0.5em;
  height: 6em;
  box-sizing: border-box;
}

.group {
  flex: 1;
  display: flex;
  align-items: center;
  justify-content: center;
  border-left: 1px solid #ece7f0;
}
.group:first-child {
  border-left: 0;
}

.item {
  margin: 0 1em;
}

.presetList {
  display: flex;
}

.preset {
  margin: 0 0.5em;
}

.preset button {
  padding: 0.7em 1em;
  color: #fff;
  font-weight: bold;
  background: #39354b;
  cursor: pointer;
  border: 0;
}
.preset button:hover, .preset button:focus, .preset.presetActive button {
  background: #a6cf39;
}

.size {
}

.sizeInput {
  display: flex;
  justify-content: center;
  align-items: center;
}

.sizeInput > div {
  margin: 0 0.5em;
}

.sizeInput input, .zoom input {
  width: 5em;
  text-align: right;
  padding: 0.5em 0 0.5em 0.5em;
  margin: 0 0.5em;
  background: none;
  border: 0;
}

.sizeInput input:hover, .zoom input:hover {
  background: #eee;
}

.zoom {
}

.zoomItem {
  min-height: 1em;
  text-align: center;
}

.zoomSlider {
  margin-top: 1em;
}
