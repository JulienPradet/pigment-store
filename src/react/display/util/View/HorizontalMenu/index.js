import React from 'react'
import styles from './horizontalmenu.css'

export const Container = ({children}) => <div className={styles.container}>{children}</div>

export const Group = ({children}) => <div className={styles.group}>{children}</div>

export const Item = ({children}) => <div className={styles.item}>{children}</div>
