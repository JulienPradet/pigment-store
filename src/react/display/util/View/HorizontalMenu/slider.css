:global .rc-slider {
  position: relative;
  height: 4px;
  width: 100%;
  border-radius: 6px;
  background-color: #e9e9e9;
  box-sizing: border-box;
  -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
}
:global .rc-slider * {
  box-sizing: border-box;
  -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
}
:global .rc-slider-track {
  position: absolute;
  left: 0;
  height: 4px;
  border-radius: 6px;
  background-color: #a6cf39;
  z-index: 1;
}
:global .rc-slider-handle {
  position: absolute;
  margin-left: -10px;
  margin-top: -8px;
  width: 20px;
  height: 20px;
  cursor: pointer;
  border-radius: 50%;
  border: solid 2px #a6cf39;
  background-color: #fff;
  z-index: 2;
}
:global .rc-slider-handle:hover {
  border-color: #a6cf39;
}
:global .rc-slider-handle-active:active {
  border-color: #a6cf39;
  box-shadow: 0 0 5px #a6cf39;
}
:global .rc-slider-mark {
  position: absolute;
  top: 10px;
  left: 0;
  width: 100%;
  font-size: 12px;
  z-index: 3;
}
:global .rc-slider-mark-text {
  position: absolute;
  display: inline-block;
  vertical-align: middle;
  text-align: center;
  cursor: pointer;
  color: #999;
}
:global .rc-slider-mark-text-active {
  color: #666;
}
:global .rc-slider-step {
  position: absolute;
  width: 100%;
  height: 4px;
  background: transparent;
  z-index: 1;
}
:global .rc-slider-dot {
  position: absolute;
  bottom: -2px;
  margin-left: -4px;
  width: 8px;
  height: 8px;
  border: 2px solid #e9e9e9;
  background-color: #fff;
  cursor: pointer;
  border-radius: 50%;
  vertical-align: middle;
}
:global .rc-slider-dot:first-child {
  margin-left: -4px;
}
:global .rc-slider-dot:last-child {
  margin-left: -4px;
}
:global .rc-slider-dot-active {
  border-color: #a6cf39;
}
:global .rc-slider-disabled {
  background-color: #e9e9e9;
}
:global .rc-slider-disabled .rc-slider-track {
  background-color: #ccc;
}
:global .rc-slider-disabled .rc-slider-handle,
:global .rc-slider-disabled .rc-slider-dot {
  border-color: #ccc;
  background-color: #fff;
  cursor: not-allowed;
}
:global .rc-slider-disabled .rc-slider-mark-text,
:global .rc-slider-disabled .rc-slider-dot {
  cursor: not-allowed !important;
}
:global .rc-slider-vertical {
  width: 4px;
  height: 100%;
}
:global .rc-slider-vertical .rc-slider-track {
  bottom: 0;
  width: 4px;
}
:global .rc-slider-vertical .rc-slider-handle {
  position: absolute;
  margin-left: -5px;
  margin-bottom: -7px;
}
:global .rc-slider-vertical .rc-slider-mark {
  top: 0;
  left: 10px;
  height: 100%;
}
:global .rc-slider-vertical .rc-slider-step {
  height: 100%;
  width: 4px;
}
:global .rc-slider-vertical .rc-slider-dot {
  left: 2px;
  margin-bottom: -4px;
}
:global .rc-slider-vertical .rc-slider-dot:first-child {
  margin-bottom: -4px;
}
:global .rc-slider-vertical .rc-slider-dot:last-child {
  margin-bottom: -4px;
}
:global .rc-slider-tooltip-zoom-down-enter,
:global .rc-slider-tooltip-zoom-down-appear {
  -webkit-animation-duration: .3s;
          animation-duration: .3s;
  -webkit-animation-fill-mode: both;
          animation-fill-mode: both;
  display: block !important;
  -webkit-animation-play-state: paused;
          animation-play-state: paused;
}
:global .rc-slider-tooltip-zoom-down-leave {
  -webkit-animation-duration: .3s;
          animation-duration: .3s;
  -webkit-animation-fill-mode: both;
          animation-fill-mode: both;
  display: block !important;
  -webkit-animation-play-state: paused;
          animation-play-state: paused;
}
:global .rc-slider-tooltip-zoom-down-enter.rc-slider-tooltip-zoom-down-enter-active,
:global .rc-slider-tooltip-zoom-down-appear.rc-slider-tooltip-zoom-down-appear-active {
  -webkit-animation-name: rcSliderTooltipZoomDownIn;
          animation-name: rcSliderTooltipZoomDownIn;
  -webkit-animation-play-state: running;
          animation-play-state: running;
}
:global .rc-slider-tooltip-zoom-down-leave.rc-slider-tooltip-zoom-down-leave-active {
  -webkit-animation-name: rcSliderTooltipZoomDownOut;
          animation-name: rcSliderTooltipZoomDownOut;
  -webkit-animation-play-state: running;
          animation-play-state: running;
}
:global .rc-slider-tooltip-zoom-down-enter,
:global .rc-slider-tooltip-zoom-down-appear {
  -webkit-transform: scale(0, 0);
          transform: scale(0, 0);
  -webkit-animation-timing-function: cubic-bezier(0.23, 1, 0.32, 1);
          animation-timing-function: cubic-bezier(0.23, 1, 0.32, 1);
}
:global .rc-slider-tooltip-zoom-down-leave {
  -webkit-animation-timing-function: cubic-bezier(0.755, 0.05, 0.855, 0.06);
          animation-timing-function: cubic-bezier(0.755, 0.05, 0.855, 0.06);
}
@-webkit-keyframes :global(rcSliderTooltipZoomDownIn) {
  0% {
    opacity: 0;
    -webkit-transform-origin: 50% 100%;
            transform-origin: 50% 100%;
    -webkit-transform: scale(0, 0);
            transform: scale(0, 0);
  }
  100% {
    -webkit-transform-origin: 50% 100%;
            transform-origin: 50% 100%;
    -webkit-transform: scale(1, 1);
            transform: scale(1, 1);
  }
}
@keyframes :global(rcSliderTooltipZoomDownIn) {
  0% {
    opacity: 0;
    -webkit-transform-origin: 50% 100%;
            transform-origin: 50% 100%;
    -webkit-transform: scale(0, 0);
            transform: scale(0, 0);
  }
  100% {
    -webkit-transform-origin: 50% 100%;
            transform-origin: 50% 100%;
    -webkit-transform: scale(1, 1);
            transform: scale(1, 1);
  }
}
@-webkit-keyframes :global(rcSliderTooltipZoomDownOut) {
  0% {
    -webkit-transform-origin: 50% 100%;
            transform-origin: 50% 100%;
    -webkit-transform: scale(1, 1);
            transform: scale(1, 1);
  }
  100% {
    opacity: 0;
    -webkit-transform-origin: 50% 100%;
            transform-origin: 50% 100%;
    -webkit-transform: scale(0, 0);
            transform: scale(0, 0);
  }
}
@keyframes :global(rcSliderTooltipZoomDownOut) {
  0% {
    -webkit-transform-origin: 50% 100%;
            transform-origin: 50% 100%;
    -webkit-transform: scale(1, 1);
            transform: scale(1, 1);
  }
  100% {
    opacity: 0;
    -webkit-transform-origin: 50% 100%;
            transform-origin: 50% 100%;
    -webkit-transform: scale(0, 0);
            transform: scale(0, 0);
  }
}
