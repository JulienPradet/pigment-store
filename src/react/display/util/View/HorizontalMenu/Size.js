import React from 'react'
import {compose, lifecycle, withState, withHandlers} from 'recompose'
import Switch from 'rc-switch'
import './switch.css'
import styles from './horizontalmenu.css'

const SizeInput = compose(
  withState('auto', 'setAuto', (props) => props.value === 'auto'),
  withState('lastValue', 'setLastValue', (props) => props.value === 'auto' ? 0 : props.value),
  lifecycle({
    componentWillReceiveProps (props) {
      if ((props.auto && props.value === 'auto') || (!props.auto && props.value !== 'auto')) {
        return
      }

      props.setAuto(props.value === 'auto')
      if (props.value !== 'auto') {
        props.setLastValue(props.value)
      }
    }
  }),
  withHandlers({
    toggle: ({name, value, onChange, lastValue, setLastValue, auto, setAuto}) => (active) => {
      if (!active) {
        setAuto(true)
        setLastValue(value)
        onChange({[name]: 'auto'})
      } else {
        setAuto(false)
        onChange({[name]: lastValue})
      }
    },
    onChange: ({name, onChange, setLastValue, auto}) => (e) => {
      if (!auto) {
        setLastValue(e.target.value)
        onChange({[name]: e.target.value})
      }
    }
  })
)(({label, name, value, lastValue, toggle, onChange, auto}) => <div className={styles.sizeInput}>
  <div>
    <Switch checked={!auto} onChange={toggle} />
  </div>
  <div>
    {label}
  </div>
  <div>
    <input type='number' value={value === 'auto' ? lastValue : value} onChange={onChange} disabled={auto} />
    px
  </div>
</div>)

const Size = (
  withHandlers({
    onChange: ({size, onChange}) => (newSize) => onChange({size: Object.assign({}, size, newSize)})
  })
)(({size, onChange}) => <div className={styles.size}>
  <SizeInput label='Width' name='width' value={size.width} onChange={onChange} />
  <SizeInput label='Height' name='height' value={size.height} onChange={onChange} />
</div>)

Size.displayName = 'Size'

export default Size
