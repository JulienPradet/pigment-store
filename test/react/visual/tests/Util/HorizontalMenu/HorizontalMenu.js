import React from 'react'
import describe from 'pigment-store/react/describe'
import {Container, Group, Item} from '../../../../../../src/core/display/App/util/View/HorizontalMenu/index'

export default describe('HorizontalMenu', Container)
  .featureJsx(
    'Default',
    () => (
      <Container>
        <Group>
          <Item>Item 1</Item>
        </Group>
        <Group>
          <Item>Item 2</Item>
        </Group>
        <Group>
          <Item>Item 3</Item>
        </Group>
      </Container>
    )
  )
