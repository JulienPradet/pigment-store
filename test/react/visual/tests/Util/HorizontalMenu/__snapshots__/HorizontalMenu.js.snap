exports[`HorizontalMenu Default 1`] = `
<div
  className="container">
  <div
    className="group">
    <div
      className="item">
      Item 1
    </div>
  </div>
  <div
    className="group">
    <div
      className="item">
      Item 2
    </div>
  </div>
  <div
    className="group">
    <div
      className="item">
      Item 3
    </div>
  </div>
</div>
`;
